Merge Sort

UserPOE-Frk opened the pull request:
Added merge sort algorithm (package name fixed).

Commit:
mergesort algorithm

Commit:
fixed package name

cegerard commented on mergesort.java:
```diff
@@ -0,0 +1,63 @@
+public class MyMergeSort {
```
This class must implements the ISortingAlgorithm interface

cegerard commented on mergesort.java:
```diff
@@ -0,0 +1,63 @@
+    private int[] tempMergArr;
+    private int length;
+ 
+    public static void main(String a[]){
```
Get rid of this method, it is not used in this project.

cegerard reviewed (changes requested):
Add the algorithm package on top of this file.

`package sortBench.algorithm`

cegerard reviewed (changes requested):
Move this file into the right directory.

src/main/java/sortBench/algorithms/merge

Commit:
package name fixed to sortBench.algorithm ;
implementing fixed ;
useless main algorithm taken off

ghost commented:
Got it all fixed !
(java path + package name + implementing + useless stuff)

Should I open another request or is the pull request keeping itself up-to-date along with my new fork ?
(Sorry. Still having a better feeling with going through my GitHub fork somehow.)

cegerard commented:
@UserPOE-Frk same comment as this PR https://github.com/cegerard/sortBench/pull/12

Commit:
Fixed class name.
Prepared file name for getting past Windows issues through direct conversion into CamelCase.

Commit:
Fixed file name to CamelCase, avoiding Windows issues.

ghost commented:
Got it into CamelCase ! Guess it should be fine now. (My bad... I do feel that Java logic is not getting into my head anytime soon !)

cegerard commented on src/main/java/sortBench/algorithms/MergeSort.java:
```diff
@@ -0,0 +1,50 @@
+public class MergeSort implements ISortingAlgorithm {
+     
+    public void sort(int inputArr[]) {
+        this.array = inputArr;
```
Ok test are failing because you need to define class attributes.
All variables prefix by `this.` are class attributes.

Below the third line (class definition) add simply lines below:
```
private int[] array;
private int length;
private int[] tempMergArr;
```

UserPOE-Frk replied:
Say I haven't been figuring it out !
Well, updated anyway. Is it going fine now ?

cegerard replied:
I did not see your change. Did you push it ?

cegerard replied:
You may have some issues with your fork history there is a lot of merge between your branches.
![image](https://user-images.githubusercontent.com/3628660/59716074-8a134d00-9215-11e9-98ec-b59e8f611cc1.png)

UserPOE-Frk replied:
I've got it as commit 9c330e4.
Kinda weird... Trying to reupload it.

I'm all okay with the history. That's but there's been a lot of differences in concept between what Git users call "master" and what I do... Hence somewhat of a fusion as if into a 'local master'. (Which wasn't much use anyway, except for Explorer-related purposes, so I just ended up leaving them as I dived down further into those separate fixes.)

UserPOE-Frk replied:
It should be okay now.

Commit:
Added class base attributes.

cegerard commented:
It is all ok now. You can merge your PR on GitHub.

ghost commented:
Done ! (Whoo !)
